src/utils/usedAndSelectedResources.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ext } from "../extensionVariables";
+
+function nodeHasIdAndMaybeResource(node: unknown): node is { id: string, resource?: { kind?: string } } {
```

Comment:
If there's a better way to be doing this, please let me know.

I couldn't find any types that exposed the `resource` property on nodes. I also found that `AzExtTreeItem` does expose the `id` property at least, but none of the nodes I tested against were able to pass the `isAzExtTreeItem` type guard so 🤷‍♀️.

Reply:
I think the type you're looking for is `AzureResourceItem`

Reply:
Yeah, if we're only looking for resources not including subs, RGs, and nodes below resources then `AzureResourceItem` should work.